source/contracts/reporting/Market.sol

```diff
@@ -136,27 +132,29 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
         return true;
     }
 
-    function disputeAutomatedReport() public returns (bool) {
-        // intentionally does not migrate the market as automated report markets won't actually migrate unless a dispute bond has been placed or the automated report doesn't occur
+    function disputeAutomatedReport() public triggersMigration returns (bool) {
         require(getReportingState() == ReportingState.AUTOMATED_DISPUTE);
-        automatedReporterDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT, tentativeWinningPayoutDistributionHash);
-        reportingWindow.getReputationToken().trustedTransfer(msg.sender, automatedReporterDisputeBondToken, AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT);
+        automatedReporterDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, Reporting.automatedReporterDisputeBondAmount(), tentativeWinningPayoutDistributionHash);
+        reportingWindow.getReputationToken().trustedTransfer(msg.sender, automatedReporterDisputeBondToken, Reporting.automatedReporterDisputeBondAmount());
         reportingWindow.updateMarketPhase();
         return true;
     }
 
     function disputeLimitedReporters() public triggersMigration returns (bool) {
         require(getReportingState() == ReportingState.LIMITED_DISPUTE);
-        limitedReportersDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT, tentativeWinningPayoutDistributionHash);
-        reportingWindow.getReputationToken().trustedTransfer(msg.sender, limitedReportersDisputeBondToken, LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT);
+        limitedReportersDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, Reporting.limitedReportersDisputeBondAmount(), tentativeWinningPayoutDistributionHash);
+        reportingWindow.getReputationToken().trustedTransfer(msg.sender, limitedReportersDisputeBondToken, Reporting.limitedReportersDisputeBondAmount());
         IReportingWindow _newReportingWindow = getBranch().getNextReportingWindow();
+        if (secondPlaceTentativeWinningPayoutDistributionHash != bytes32(0)) {
+            updateTentativeWinningPayoutDistributionHash(secondPlaceTentativeWinningPayoutDistributionHash);
+        }
```

Comment:
Shouldn't this same block be added to the other two dispute functions?

Reply:
For `automated` there can't be a second place and for `all` the payoutHash is no longer relevant, but we could do this anyway in both with no negative side effects except gas cost.

Reply:
Actually, I think no matter `updateTentativeWinningPayoutDistributionHash` should be called.  It should then contain all of the logic necessary to figure out the top 2 winners based on provided hash and existing first and second place (which can be fetched from storage in that function).  So I would get rid of this if block and just call `updateTentativeWinningPayoutDistributionHash(disputedHash)`.

Reply:
I would prefer to just always call `updateTentativeWinningPayoutDistributionHash` for clarity, even if unnecessary.  We should be able to throw some early-exit checks at the top that make it so most of the function isn't run if the three values are already correctly "sorted".  I believe I see what you are saying with being able to avoid the code path in some situations, but I'm generally hesitant to rely on functions being called in certain contexts in order for them to behave correctly.

Reply:
Will do then! If we have to optimize in some way later I don't think it'll be hard to figure out.